File: src/main/java/ting/leetcode/java/arrays101/TwoSum2.java
package ting.leetcode.java.arrays101;

public class TwoSum2 {
    public int[] twoSum(int[] numbers, int target) {
        int left = 0;
        int right = numbers.length - 1;

        while (numbers[left] + numbers[right] != target) {
            int sum = numbers[left] + numbers[right];
            if (sum > target) {
                right--;
            } else if (sum < target) {
                left++;
            }
        }

        return new int[] { left + 1, right + 1 };
    }
}

File: src/main/kotlin/ting/leetcode/arrayandstring/TwoSum2.kt
package ting.leetcode.arrayandstring

class TwoSum2 {
    fun twoSum(numbers: IntArray, target: Int): IntArray {
        val result = IntArray(2)
        numbers.forEachIndexed { i, firstValue ->
            for (j in (i + 1)..numbers.lastIndex) {
                if (firstValue + numbers[j] == target) {
                    result[0] = i + 1
                    result[1] = j + 1
                    return result
                }
            }
        }
        return result
    }

    fun twoSumWithTwoPinter(numbers: IntArray, target: Int): IntArray {
        var firstPointer = 0
        var secondPointer = numbers.lastIndex
        while (numbers[firstPointer] + numbers[secondPointer] != target) {
            if (numbers[firstPointer] + numbers[secondPointer] > target) secondPointer-- else firstPointer++
        }
        return intArrayOf(++firstPointer, ++secondPointer)
    }
}